File: 114/114_version2.java
class Solution {
    public void flatten(TreeNode root) {
       TreeNode current = root;
        while(current!=null){
            if(current.left!=null){
                TreeNode tempLeft = current.left;
                while(tempLeft.right!=null) tempLeft = tempLeft.right;
                tempLeft.right = current.right;
                current.right = current.left;
                current.left = null;
            }
            current = current.right;
        }
    }
}

File: 114/114_version2.py
class Solution:
    def flatten(self, root: TreeNode) -> None:
        """
        Do not return anything, modify root in-place instead.
        """
        current = root
        while current:
            if current.left:
                temp = current.left
                while temp.right:
                    temp = temp.right
                temp.right = current.right
                current.right = current.left
                current.left = None
            current = current.right
